proposals/simd/portable-simd.md

```diff
@@ -0,0 +1,1320 @@
+```
+
+## Floating-point sign bit operations
+
+These floating point operations are simple manipulations of the sign bit. No
+changes are made to the exponent or trailing significand bits, even for NaN
+inputs.
+
+### Negation
+* `f32x4.neg(a: v128) -> v128`
+* `f64x2.neg(a: v128) -> v128`
+
+Apply the IEEE `negate(x)` function to each lane. This simply inverts the sign
+bit, preserving all other bits.
+
+```python
+def S.neg(a):
+    return S.lanewise_unary(ieee.negate, a)
+```
+
+### Absolute value
+* `f32x4.abs(a: v128) -> v128`
+* `f64x2.abs(a: v128) -> v128`
+
+Apply the IEEE `abs(x)` function to each lane. This simply clears the sign bit,
+preserving all other bits.
+
+```python
+def S.abs(a):
+    return S.lanewise_unary(ieee.abs, a)
+```
+
+## Floating-point min and max
+
+These operations are not part of the IEEE 754-2008 standard. Notably, the
+`minNum` and `maxNum` operations defined here behave differently than the IEEE
+`minNum` and `maxNum` operations when one operand is a signaling NaN.
+
+The minimum and maximum value of +0 and -0 is computed as if -0 < +0.
+
+### NaN-propagating minimum
+* `f32x4.min(a: v128, b: v128) -> v128`
+* `f64x2.min(a: v128, b: v128) -> v128`
+
+Lane-wise minimum value, propagating NaNs:
+
+```python
+def S.min(a, b):
+    def min(x, y):
+        if isnan(x) or isnan(y):
+            return propagate_nan(x, y)
+        # Prefer -0 for min(-0, +0) and min(+0, -0).
+        if x == 0 and y == 0 and signbit(x) != signbit(y):
+            return -0.0
+        if x < y:
+            return x
+        else:
+            return y
+    return S.lanewise_binary(min, a, b)
+```
+
+### NaN-propagating maximum
+* `f32x4.max(a: v128, b: v128) -> v128`
+* `f64x2.max(a: v128, b: v128) -> v128`
+
+Lane-wise maximum value, propagating NaNs:
+
+```python
+def S.max(a, b):
+    def max(x, y):
+        if isnan(x) or isnan(y):
+            return propagate_nan(x, y)
+        # Prefer +0 for max(-0, +0) and max(+0, -0).
+        if x == 0 and y == 0 and signbit(x) != signbit(y):
+            return +0.0
+        if x > y:
+            return x
+        else:
+            return y
+    return S.lanewise_binary(max, a, b)
+```
+
+### NaN-suppressing minimum
+* `f32x4.minNum(a: v128, b: v128) -> v128`
+* `f64x2.minNum(a: v128, b: v128) -> v128`
+
+Lane-wise minimum value, suppressing single NaNs:
+
+```python
+def S.minNum(a, b):
+    def minNum(x, y):
+        if isnan(x) and isnan(y):
+            return propagate_nan(x, y)
+        if isnan(x):
+            return y
+        if isnan(y):
+            return x
+        # Prefer -0 for min(-0, +0) and min(+0, -0).
+        if x == 0 and y == 0 and signbit(x) != signbit(y):
+            return -0.0
+        if x < y:
+            return x
+        else:
+            return y
+    return S.lanewise_binary(minNum, a, b)
+```
+
+Note that this function behaves differently than the IEEE 754 `minNum` function
+when one of the operands is a signaling NaN.
```

Comment:
Seems like it was agreed upon in Simd.js to remove minNum and maxNum due to the compliance issue and not enough use cases. https://github.com/tc39/ecmascript_simd/issues/341
Does it make sense to remove it from portable_simd as well?
And the replacement from IEEE754-2018 can be introduced to wasm first if it seems fit when it lands.